lib/Bio/EnsEMBL/DataCheck/Checks/CheckMartDatasetName.pm

```diff
@@ -0,0 +1,100 @@
+=head1 LICENSE
+
+Copyright [2018-2019] EMBL-European Bioinformatics Institute
+
+Licensed under the Apache License, Version 2.0 (the 'License');
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an 'AS IS' BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=cut
+
+package Bio::EnsEMBL::DataCheck::Checks::CheckMartDatasetName;
+
+use warnings;
+use strict;
+
+use Moose;
+use Test::More;
+use Bio::EnsEMBL::DataCheck::Test::DataCheck;
+use MartUtils qw(generate_dataset_name_from_db_name);
+use Bio::EnsEMBL::BioMart::Mart qw(genome_to_include);
+use Bio::EnsEMBL::DataCheck::Utils qw/repo_location/;
+use Bio::EnsEMBL::MetaData::Base qw(fetch_and_set_release);
+
+extends 'Bio::EnsEMBL::DataCheck::DbCheck';
+
+use constant {
+  NAME           => 'CheckMartDatasetName',
+  DESCRIPTION    => 'Check that the given core database mart dataset name is not exceeding MySQL limit of 64 chars and that they are unique accross all the divisions (except Bacteria)',
+  GROUPS         => ['core','biomart'],
+  DATACHECK_TYPE => 'critical',
+  DB_TYPES       => ['core'],
+  TABLES         => ['meta']
+};
+
+sub tests {
+  my ($self) = @_;
+  SKIP: {
+    my $dbname = $self->dba->dbc->dbname;
+    my $mart_dataset = generate_dataset_name_from_db_name($dbname);
+    my $mca = $self->dba->get_adaptor('MetaContainer');
+    my $division = $mca->get_division;
+    my $production_name = $mca->get_production_name;
+    my $schema_version = $mca->get_schema_version;
+    if ($division eq "EnsemblVertebrates"){
+      my $repo_location = repo_location('ensembl-biomart');
+      $repo_location =~ s/ensembl-biomart//;
+      # Load species to include in the Vertebrates marts
+      my $included_species = genome_to_include($division,$repo_location);
+      if (!grep( /$production_name/, @$included_species) ){
+        skip 'We dont build mart for this species', 1;
+      }
+    }
+    # If dataset is longer than 18 char, we won't be able to generate gene mart tables
+    # like dataset_gene_ensembl__protein_feature_superfamily__dm as this will exceed
+    # the MySQL table name limit of 64 char.
+    if (length($mart_dataset) > 18) {
+      fail("$mart_dataset name is too long. Check with data team and remove this species $dbname from the include ini file\n");
+      return;
+    }
+    my $metadata_dba = $self->registry->get_DBAdaptor('multi', 'metadata');
```

Comment:
It's best to use the DbCheck `get_dba` method for this; as well as looking a bit neater, that ensures that the db connection will get properly cleared up when the datacheck completes.
```suggestion
    my $metadata_dba = $self->get_dba('multi', 'metadata');
```

Reply:
Sure, thanks for letting me know about this.